editors/emacs/ra-emacs-lsp.el

```diff
@@ -0,0 +1,139 @@
+;;; ra-emacs-lsp.el --- Rust analyzer emacs bindings for emacs-lsp -*- lexical-binding: t; -*-
+;;; Code:
+
+(require 'lsp)
+(require 'dash)
+(require 'ht)
+
+;; This currently
+;;  - sets up rust-analyzer with emacs-lsp, giving
+;;    - code actions
+;;    - completion (use company-lsp for proper snippet support)
+;;    - imenu support
+;;    - on-type formatting
+;;    - 'hover' type information & documentation (with lsp-ui)
+;;  - implements source changes (for code actions etc.), except for file system changes
+;;  - implements joinLines (you need to bind rust-analyzer-join-lines to a key)
+;;  - implements extendSelection (either bind rust-analyzer-extend-selection to a key, or use expand-region)
+
+;; What's missing:
+;;  - file system changes in apply-source-change
+;;  - semantic highlighting
+;;  - onEnter, parentModule, findMatchingBrace
+;;  - runnables
```

Comment:
fwiw, I find this one pretty useful for coding: being able to run a single test without much ceremony is cool!

Reply:
Yeah, I currently use https://github.com/kwrooijen/cargo.el, which allows me to run all tests for the current crate at least (though that's more by accident because it doesn't handle workspaces, I think...); it theoretically also has support for running single tests, but that's somehow broken. I should add support for that and get rid of cargo.el :smile: